test/onnx/test_onnx_export.py

```diff
@@ -19,27 +20,66 @@
 sys.path.append(pytorch_test_dir)
 
 
+def export_to_onnx(
+    model: Union[torch.nn.Module, torch.jit.ScriptFunction],
+    input: Tuple[torch.Tensor],
+    custom_ops: Optional[List[Tuple[str, Callable, int]]] = None,
+    mocks: Optional[List[unittest.mock.patch]] = None,
+    operator_export_type: OperatorExportTypes = OperatorExportTypes.ONNX,
+    opset_version=torch.onnx.symbolic_helper._export_onnx_opset_version,
+):
+    """Exports `model(input)` to ONNX and return it.
+
+    Custom operators and/or unittest patches can be used help reproducing specific behaviors.
+
+    Args:
+        model: model to export
+        input: model input with same format as `torch.onnx.export(..,args,...)`
+        custom_ops: list of custom operators to use during export
+        mocks: list of unittests mocks to use during export
+        operator_export_type: export type as described by `torch.onnx.export(...operator_export_type,...)`
+        opset_version: ONNX opset version as described by `torch.onnx.export(...opset_version,...)`
+    Returns:
+        A valid ONNX model (`onnx.ModelProto`)
+    """
+
+    custom_ops = custom_ops or []
+    mocks = mocks or []
+    with ExitStack() as stack:
+        for mgr in [*custom_ops, *mocks]:
```

Comment:
Curious: Are custom_ops context managers? How does this work?

Reply:
It can be anything that implements `__enter__` and `__exit__` but I will restrict it to context manager/decorators